openapi/openapi-client/pom.xml

```diff
@@ -197,6 +201,70 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <configuration>
+          <relocations>
+            <relocation>
+              <pattern>io.swagger</pattern>
+              <shadedPattern>${shading.prefix}.io.swagger</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>org.openapitools</pattern>
+              <shadedPattern>${shading.prefix}.org.openapitools</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>com.google</pattern>
+              <shadedPattern>${shading.prefix}.com.google</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>javax.annotation</pattern>
+              <shadedPattern>${shading.prefix}.javax.annotation</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>org.glassfish.jersey</pattern>
+              <shadedPattern>${shading.prefix}.org.glassfish.jersey</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>com.fasterxml.jackson</pattern>
+              <shadedPattern>${shading.prefix}.com.fasterxml.jackson</shadedPattern>
+            </relocation>
+          </relocations>
+          <artifactSet>
+            <includes>
+              <include>io.swagger:swagger-annotations</include>
+              <include>io.swagger:swagger-models</include>
+              <include>org.openapitools:jackson-databind-nullable</include>
+              <include>com.google.code.findbugs:jsr305</include>
+              <include>org.glassfish.jersey.core:jersey-client</include>
+              <include>org.glassfish.jersey.media:jersey-media-json-jackson</include>
+              <include>com.fasterxml.jackson.core:jackson-annotations</include>
+              <include>com.fasterxml.jackson.core:jackson-databind</include>
+              <include>com.fasterxml.jackson.datatype:jackson-datatype-jsr310</include>
+              <include>org.glassfish.jersey.media:jersey-media-multipart</include>
+            </includes>
+          </artifactSet>
+          <filters>
+            <filter>
+              <artifact>*:*</artifact>
+              <excludes>
+                <exclude>META-INF/*.SF</exclude>
+                <exclude>META-INF/*.DSA</exclude>
+                <exclude>META-INF/*.RSA</exclude>
+                <exclude>**/log4j.properties</exclude>
+                <exclude>META-INF/LICENSE.txt</exclude>
+                <exclude>META-INF/NOTICE.txt</exclude>
+                <exclude>LICENSE.txt</exclude>
+                <exclude>NOTICE.txt</exclude>
+              </excludes>
+            </filter>
+          </filters>
+          <transformers>
+            <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"></transformer>
```

Comment:
Does the SBT also perform the same service resource transformation?

Reply:
Per the result, they are consistent:

For mvn:
```
(base) ➜  celeborn git:(openapi_client_shade) jar -tf openapi/openapi-client/target/celeborn-openapi-client_2.12-0.6.0-SNAPSHOT.jar|grep -v celeborn
META-INF/
META-INF/MANIFEST.MF
org/
org/apache/
META-INF/NOTICE
META-INF/maven/
META-INF/DEPENDENCIES
META-INF/LICENSE
META-INF/maven/io.swagger/
META-INF/maven/io.swagger/swagger-annotations/
META-INF/maven/io.swagger/swagger-annotations/pom.xml
META-INF/maven/io.swagger/swagger-annotations/pom.properties
META-INF/services/
META-INF/maven/org.openapitools/
META-INF/maven/org.openapitools/jackson-databind-nullable/
META-INF/maven/org.openapitools/jackson-databind-nullable/pom.xml
META-INF/maven/org.openapitools/jackson-databind-nullable/pom.properties
META-INF/LICENSE.md
META-INF/NOTICE.md
META-INF/maven/org.glassfish.jersey.core/
META-INF/maven/org.glassfish.jersey.core/jersey-client/
META-INF/maven/org.glassfish.jersey.core/jersey-client/pom.properties
META-INF/maven/org.glassfish.jersey.core/jersey-client/pom.xml
META-INF/native-image/
META-INF/native-image/org.glassfish.jersey.core/
META-INF/native-image/org.glassfish.jersey.core/jersey-client/
META-INF/native-image/org.glassfish.jersey.core/jersey-client/reflect-config.json
META-INF/native-image/org.glassfish.jersey.core/jersey-client/resource-config.json
META-INF/NOTICE.markdown
META-INF/maven/org.glassfish.jersey.media/
META-INF/maven/org.glassfish.jersey.media/jersey-media-json-jackson/
META-INF/maven/org.glassfish.jersey.media/jersey-media-json-jackson/pom.properties
META-INF/maven/org.glassfish.jersey.media/jersey-media-json-jackson/pom.xml
META-INF/maven/com.fasterxml.jackson.core/
META-INF/maven/com.fasterxml.jackson.core/jackson-annotations/
META-INF/maven/com.fasterxml.jackson.core/jackson-annotations/pom.xml
META-INF/maven/com.fasterxml.jackson.core/jackson-annotations/pom.properties
META-INF/maven/com.fasterxml.jackson.core/jackson-databind/
META-INF/maven/com.fasterxml.jackson.core/jackson-databind/pom.xml
META-INF/maven/com.fasterxml.jackson.core/jackson-databind/pom.properties
META-INF/versions/
META-INF/versions/9/
META-INF/versions/9/module-info.class
META-INF/maven/com.fasterxml.jackson.datatype/
META-INF/maven/com.fasterxml.jackson.datatype/jackson-datatype-jsr310/
META-INF/maven/com.fasterxml.jackson.datatype/jackson-datatype-jsr310/pom.properties
META-INF/maven/com.fasterxml.jackson.datatype/jackson-datatype-jsr310/pom.xml
META-INF/maven/org.glassfish.jersey.media/jersey-media-multipart/
META-INF/maven/org.glassfish.jersey.media/jersey-media-multipart/pom.properties
META-INF/maven/org.glassfish.jersey.media/jersey-media-multipart/pom.xml
META-INF/services/jakarta.validation.valueextraction.ValueExtractor
META-INF/services/javax.validation.valueextraction.ValueExtractor
(base) ➜  celeborn git:(openapi_client_shade) 

```

For sbt:
```
META-INF/
META-INF/MANIFEST.MF
META-INF/maven/
META-INF/maven/com.fasterxml.jackson.core/
META-INF/maven/com.fasterxml.jackson.core/jackson-annotations/
META-INF/maven/com.fasterxml.jackson.core/jackson-annotations/pom.properties
META-INF/maven/com.fasterxml.jackson.core/jackson-annotations/pom.xml
META-INF/maven/com.fasterxml.jackson.core/jackson-databind/
META-INF/maven/com.fasterxml.jackson.core/jackson-databind/pom.properties
META-INF/maven/com.fasterxml.jackson.core/jackson-databind/pom.xml
META-INF/maven/com.fasterxml.jackson.datatype/
META-INF/maven/com.fasterxml.jackson.datatype/jackson-datatype-jsr310/
META-INF/maven/com.fasterxml.jackson.datatype/jackson-datatype-jsr310/pom.properties
META-INF/maven/com.fasterxml.jackson.datatype/jackson-datatype-jsr310/pom.xml
META-INF/maven/io.swagger/
META-INF/maven/io.swagger/swagger-annotations/
META-INF/maven/io.swagger/swagger-annotations/pom.properties
META-INF/maven/io.swagger/swagger-annotations/pom.xml
META-INF/maven/io.swagger/swagger-models/
META-INF/maven/io.swagger/swagger-models/pom.properties
META-INF/maven/io.swagger/swagger-models/pom.xml
META-INF/maven/org.glassfish.jersey.core/
META-INF/maven/org.glassfish.jersey.core/jersey-client/
META-INF/maven/org.glassfish.jersey.core/jersey-client/pom.properties
META-INF/maven/org.glassfish.jersey.core/jersey-client/pom.xml
META-INF/maven/org.glassfish.jersey.media/
META-INF/maven/org.glassfish.jersey.media/jersey-media-json-jackson/
META-INF/maven/org.glassfish.jersey.media/jersey-media-json-jackson/pom.properties
META-INF/maven/org.glassfish.jersey.media/jersey-media-json-jackson/pom.xml
META-INF/maven/org.glassfish.jersey.media/jersey-media-multipart/
META-INF/maven/org.glassfish.jersey.media/jersey-media-multipart/pom.properties
META-INF/maven/org.glassfish.jersey.media/jersey-media-multipart/pom.xml
META-INF/maven/org.openapitools/
META-INF/maven/org.openapitools/jackson-databind-nullable/
META-INF/maven/org.openapitools/jackson-databind-nullable/pom.properties
META-INF/maven/org.openapitools/jackson-databind-nullable/pom.xml
META-INF/native-image/
META-INF/native-image/org.glassfish.jersey.core/
META-INF/native-image/org.glassfish.jersey.core/jersey-client/
META-INF/native-image/org.glassfish.jersey.core/jersey-client/reflect-config.json
META-INF/native-image/org.glassfish.jersey.core/jersey-client/resource-config.json
META-INF/services/
META-INF/services/jakarta.validation.valueextraction.ValueExtractor
META-INF/services/javax.validation.valueextraction.ValueExtractor
META-INF/versions/
META-INF/versions/9/
META-INF/versions/9/module-info.class
module-info.class
org/
org/apache/
```

Reply:
The shade mvn/sbt plugin setting are the same with the existing shade modules.